nix/buildLeanPackage.nix

```diff
@@ -114,31 +146,39 @@ with builtins; let
     ${lean-emacs}/bin/emacs --eval "(progn (setq lean4-rootdir \"${lean}\") (require 'lean4-mode))" "$@"
   '';
   makeVSCodeWrapper = name: lean: writeShellScriptBin name ''
-    PATH=${lean}/bin:$PATH ${lean-vscode}/bin/code "$@"
+    PATH=${lean}/bin:$PATH LEAN_WORKER_PATH=${leanArg}/bin/lean ${lean-vscode}/bin/code "$@"
   '';
   printPaths = deps: writeShellScriptBin "print-paths" ''
     echo "${depRoot "print-paths" deps}"
-    echo ".:${lib.concatStringsSep ":" (map (dep: dep.src) (attrValues allDeps))}"
+    echo ".:${lib.concatStringsSep ":" (map (dep: dep.src) (attrValues allExternalDeps))}"
   '';
   makePrintPathsFor = deps: mods: printPaths deps // mapAttrs (_: mod: makePrintPathsFor (deps ++ [mod]) mods) mods;
-in rec {
-  inherit name lean deps allDeps print-lean-deps src;
   mods      = buildModAndDeps name {};
+  makeStaticLibDeriv = objects: runCommand "${name}-lib" { buildInputs = [ stdenv.cc.bintools.bintools ]; } ''
+    mkdir -p $out
+    ar Trcs $out/lib${name}.a ${lib.concatStringsSep " " (map (drv: "${drv}/${drv.oPath}") (attrValues objects))}
+  '';
+in rec {
+  inherit name lean deps staticLibDeps allExternalDeps print-lean-deps src mods;
   modRoot   = depRoot name [ mods.${name} ];
   cTree     = symlinkJoin { name = "${name}-cTree"; paths = map (mod: mod.c) (attrValues mods); };
   objects   = mapAttrs compileMod mods;
   oTree     = symlinkJoin { name = "${name}-oTree"; paths = (attrValues objects); };
-  staticLib = runCommand "${name}-lib" { buildInputs = [ stdenv.cc.bintools.bintools ]; } ''
-    mkdir $out
-    ar Trcs $out/lib${name}.a ${lib.concatStringsSep " " (map (drv: "${drv}/${drv.oPath}") (attrValues objects))}
+  staticLib = makeStaticLibDeriv objects;
+  plugin = runCommand "${name}.so" { buildInputs = [ stdenv.cc ]; } ''
```

Comment:
Could probably be called more generally `sharedLib`?

Reply:
The semantic distinction I had in my mind was "plugin" = "sharedLib" + compatible init function symbol. But I'm happy to change to sharedLib if you prefer that.

Reply:
There is nothing special about plugin inits compared to other packages, so calling it `sharedLib` should be fine. Let's also follow the output structure of `staticLib`, unless you see a reason not to.
We shouldn't need `-rdynamic` on libraries btw.

Reply:
Fixed.